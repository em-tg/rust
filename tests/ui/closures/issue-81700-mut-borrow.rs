//@ check-pass
fn foo(x: &mut u32) {
    let bar = || { foo(x); };
    bar(); //~ WARNING cannot borrow
}
fn main() {}
