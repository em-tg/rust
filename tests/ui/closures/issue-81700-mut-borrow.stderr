warning[E0596]: cannot borrow `bar` as mutable, as it is not declared as mutable
  --> $DIR/issue-81700-mut-borrow.rs:4:5
   |
LL |     let bar = || { foo(x); };
   |                        - calling `bar` requires mutable binding due to mutable borrow of `x`
LL |     bar();
   |     ^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut bar = || { foo(x); };
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
